proto/minder/v1/minder.proto

```diff
@@ -3348,6 +3348,12 @@ message Provider {
     // credentials_state is the state of the credentials for the provider.
     // This is an output-only field. It may be: "set", "unset", "not_applicable".
     string credentials_state = 9;
+
+    // id is the unique identifier of the provider.
+    string id = 10 [
+        (buf.validate.field).string = {uuid: true},
+        (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+    ];
```

Comment:
Not strictly required, but you might want to also add the `(google.api.field_behavior) = OUTPUT_ONLY` annotation, since I expect that we would reject changing the ID on create/update.

Here's an example:

https://github.com/mindersec/minder/blob/main/proto/minder/v1/minder.proto#L3646

Reply:
Sure will do a follow up PR to address this 
Thanks